core/src/main/java/tc/oc/pgm/structure/DynamicScheduler.java

```diff
@@ -0,0 +1,55 @@
+package tc.oc.pgm.structure;
+
+import java.util.Comparator;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchScope;
+
+/**
+ * We need this to make sure that dynamics are placed and cleared in definition order, and that
+ * clears happen before placements.
+ */
+public class DynamicScheduler {
```

Comment:
given how little work the scheduler does, and how little work the structure match module does, i'd consider just joining the logic of both together